aten/src/ATen/native/cpu/ReduceOpsKernel.cpp

```diff
@@ -112,12 +112,65 @@ static void cumprod_cpu_kernel(const Tensor& result, const Tensor& self, int64_t
     );
   });
 }
+// custom min and max to be used in logcumsumexp for complex arguments
+template <typename scalar_t>
+c10::complex<scalar_t> _logcumsumexp_minmax(c10::complex<scalar_t> x, c10::complex<scalar_t> y, bool min) {
+  scalar_t xr = std::real(x);
+  scalar_t yr = std::real(y);
+  if (std::isnan(yr) || (std::isnan(std::imag(y)))) {
```

Comment:
I'm not an expert in c10 either and I'm not sure if this function is already available. It's better to discuss it with c10 experts.

Reply:
I see no harm in implementing such template in https://github.com/pytorch/pytorch/blob/master/c10/util/complex_utils.h

Reply:
I implemented it in the newest commit. I was hesitated first because there is no agreed behaviour for `std::nan` for complex numbers, but `torch.isnan` defined a complex number to be nan if either real or imag part is nan.

Reply:
I was tempted to suggest using `at::_isnan`, but dispatch complexity would probably be an overkill. Thank you for addressing the suggested change. Please fix the lint though